### ISSUE TITLE
Cached file entries get cut off [Serving from Google Drive]

### ISSUE THREAD

**Issue by sheepmaster** [kzahel/web-server-chrome#15]
Steps to reproduce:
1. Serve a directory from Google Drive through the Files app on Chrome OS
2. Open any file from the directory over HTTP
3. Change the file by making it longer
4. Reload the file

Result:
The returned file gets cut off after the first n bytes, where n is the initial length of the file. If you change something in the first n bytes, the returned file will reflect those changes, but the file will always be cut off after n bytes.

From what I could gather by debugging, an old File object is cached in the entryFileCache and does not get invalidated when the file is changed (in contrast to the comment in EntryCache.get()). The actual file contents are still read freshly from disk, but because the old File object contains the old file size, only that number of bytes is read.

Given that the actual file contents are still read from disk, is there really much to be gained from caching the File objects at all?

**Comment by kzahel** [comment]
Thanks for looking into this. I think I had that caching mechanism in place to speed things up a bit, I can't remember even why. I think I might just invalidate the cache or disable it completely to fix this problem.

**Comment by drcmda** [comment]
Yep, this becomes a mayor stumbling block for development. It doesn't just cut, when you edit any file to a smaller size it starts to overlap with the previous one creating entities that can't be parsed by the browser. Is there progress on this one?

**Comment by kzahel** [comment]
When I use a text editor app to save files in google drive, and then serve them from the web server, they don't even update the contents. It doesn't seem to work very well at all for serving from google drive folders. Not sure if there is really anything I can do to help this.

**Comment by sheepmaster** [comment]
I think this only happens on Google Drive folders, because the modification time there might not get updated properly. Disabling the entryFileCache should fix that.

**Comment by drcmda** [comment]
I believe it even happened locally. If the file cache isn't making a large difference as you assumed maybe it should be switched off.

**Comment by drcmda** [comment]
I can confirm now that this happens even if the files are stored on the SSD. Something is destroying the served files. For instance changing a few bytes and then saving creates an overlap that results in doubled closing tags at the end of the .js file i'm editing.

});
}); <--- not in the original file

**Comment by kzahel** [comment]
I tried to reproduce the problem with a file I was editing in my downloads
folder but it never gave me any problem. I'll try again next week. All you
have to do is make the file smaller in size?

**Comment by drcmda** [comment]
With Google drive I can reproduce it in every case. Files served from the local SSD couldn't be processed when I use an IDE like Google's DevEditor. If I use Chrome itself as an IDE by using its devtools plus the local resources setting (caching off), it seems to work.

**Comment by kzahel** [comment]
Saw another bug report of this happening, on mac osX... but I am still not sure how to reproduce it.

**Comment by kzahel** [comment]
I disabled the entry file cache completely. Hopefully this fixes the issue.
